packages/node/src/utils/cosmos.ts

```diff
@@ -69,10 +70,16 @@ export function filterMessageData(
   }
   if (filter.values) {
     for (const key in filter.values) {
-      if (
-        filter.values[key] !==
-        key.split('.').reduce((acc, curr) => acc[curr], data.msg.decodedMsg)
-      ) {
+      let decodedMsgData = key
+        .split('.')
+        .reduce((acc, curr) => acc[curr], data.msg.decodedMsg);
+
+      //stringify Long for equality check
+      if (isLong(decodedMsgData)) {
+        decodedMsgData = decodedMsgData.toString();
```

Comment:
What happens if a user puts a number as the filter? I think we should compare long with string and number